packages/hooks/src/useCollectionQuery/mdxUtils.ts

```diff
@@ -0,0 +1,148 @@
+import { ApolloClient, InMemoryCache } from "@apollo/client";
+import gql from "graphql-tag";
+
+export interface ListQueryType {
+  allPlanets: {
+    __typename?: "PlanetsConnection";
+    pageInfo: {
+      hasNextPage: boolean;
+      endCursor: string;
+    };
+    edges: Array<{
+      __typename?: "PlanetsEdge";
+      node: {
+        __typename?: "Planet";
+        name: "string";
+        id: "string";
+      };
+      cursor: string;
+    }>;
+    totalCount: number;
+  };
+}
+
+export const LIST_QUERY = gql`
+  query ListQuery($cursor: String) {
+    allPlanets(first: 4, after: $cursor) {
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      edges {
+        node {
+          name
+          id
+        }
+        cursor
+      }
+      totalCount
+    }
+  }
+`;
+
+export const apolloClient = new ApolloClient({
+  uri: "https://swapi-graphql.netlify.app/.netlify/functions/index",
+  cache: new InMemoryCache(),
+});
+
+interface ListItems {
+  id?: string;
+  icon?: string;
+  content?: string;
+}
+
+export function mapDataToListItem(
+  data: ListQueryType,
+): Array<ListItems> | undefined {
+  if (data !== undefined) {
+    const items = data.allPlanets?.edges.map(edge => {
+      return {
+        id: edge.node.id,
+        icon: "starFill",
+        iconColor: "green",
+        content: edge.node.name,
+      };
+    });
+    return items;
+  }
+  return undefined;
+}
```

Comment:
I think if we use the List component we can probably get rid of this and just do a simple map inline. This is hiding away interactions that I think would be good for the consumer of the component to see and, map could handle pretty easily within the playground.

Reply:
We can totally do that. Did you know the playground does not like the `?.` syntax? However, I can just ensure edges exist before the mapping.